packages/server/src/hub/handlers/handle-wallet-message.ts

```diff
@@ -13,23 +13,33 @@ import { handleOngoingProcessAction } from './handle-ongoing-process-action';
 export async function handleWalletMessage(
   message: RelayableAction,
 ): Promise<MessageRelayRequested | undefined> {
-  if (isNewProcessAction(message)) {
+  if (isNewProcessAction(message) && (await shouldHandleAsNewProcessAction(message))) {
     return handleNewProcessAction(message);
   } else if (isProtocolAction(message)) {
-    const { processId } = message;
-    const process = await getProcess(processId);
-    if (!process) {
-      throw new Error(`Process ${processId} is not running.`);
-    }
     return handleOngoingProcessAction(message);
   } else {
     return undefined;
   }
 }
+
 // TODO: We should define types for NewProcessAction and ProtocolAction
 
-function isNewProcessAction(action: RelayableAction): action is ConcludeInstigated {
-  return action.type === 'WALLET.NEW_PROCESS.CONCLUDE_INSTIGATED';
+async function shouldHandleAsNewProcessAction(
+  action: ConcludeInstigated | CommitmentsReceived,
+): Promise<boolean> {
+  if (action.type !== 'WALLET.COMMON.COMMITMENTS_RECEIVED') {
+    return true;
+  }
+  return !(await getProcess(action.processId));
```

Comment:
The only downside of that approach is that if for some reason the client decides to broadcast commitments instead of send them serially then this check won't work.